src/main/java/seedu/exercise/logic/commands/statistic/StatsFactory.java

```diff
@@ -0,0 +1,281 @@
+    private String category;
+    private Date startDate;
+    private Date endDate;
+
+    /**
+     * Generates a StatsFactory object that can generate Statistic.
+     */
+    public StatsFactory(ReadOnlyResourceBook<Exercise> exercises, String chart, String category,
+                        Date startDate, Date endDate) {
+        this.exercises = exercises.getResourceList();
+        this.chart = chart;
+        this.category = category;
+        if (startDate == null && endDate == null) {
+            this.startDate = Date.getOneWeekBeforeToday();
+            this.endDate = Date.getToday();
+        } else {
+            this.startDate = startDate;
+            this.endDate = endDate;
+        }
+    }
+
+    /**
+     * Generates and returns statistic for different chart type.
+     */
+    public Statistic generateStatistic() {
+        HashMap<String, Double> data;
+        if (chart.equals("linechart")) {
+
+            return generateLineChartStatistic();
+
+        } else if (chart.equals("piechart")) {
+
+            return generatePieChartStatistic();
+
+        } else { //barchart
+
+            return generateBarChartStatistic();
+
+        }
+    }
+
+    /**
+     * Generate statistic for line chart.
+     */
+    private Statistic generateLineChartStatistic() {
+        ArrayList<Date> dates = Date.getListOfDates(startDate, endDate);
+        ArrayList<Double> values;
+
+        if (category.equals("exercise")) {
+            values = exerciseCountByDate(getFilteredExercise(), dates);
+        } else {
+            values = caloriesByDate(getFilteredExercise(), dates);
+        }
+
+        return new Statistic(category, chart, startDate, endDate, datesToString(dates), values);
+    }
+
+    /**
+     * Generate statistic for bar chart.
+     */
+    private Statistic generateBarChartStatistic() {
+        HashMap<String, Double> data;
+        if (category.equals("exercise")) {
+            data = getTotalExerciseQuantity();
+        } else { //calories
+            data = getTotalCaloriesData();
+        }
+
+        ArrayList<String> names = hashMapNameToList(data);
+        ArrayList<Double> values = hashMapDoubleToList(data, names);
+
+        return new Statistic(category, chart, startDate, endDate, names, values);
+    }
+
+    /**
+     * Generate statistic for pie chart.
+     */
+    private Statistic generatePieChartStatistic() {
+        HashMap<String, Double> data;
+        if (category.equals("exercise")) {
+            data = getTotalExerciseCount();
+        } else { //calories
+            data = getTotalCaloriesData();
+        }
+
+        ArrayList<String> names = hashMapNameToList(data);
+        ArrayList<Double> values = hashMapDoubleToList(data, names);
+
+        return new Statistic(category, chart, startDate, endDate, names, values);
+    }
+
+    /**
+     * Compute exercise count with filtered exercises list.
+     */
+    private HashMap<String, Double> getTotalExerciseCount() {
```

Comment:
Maybe you can just return a `list` of `double` and abstract out the methods. The `getTotalExerciseCount()` is very similar to `getTotalExerciseQuantity()`. Entirely up to you if you wish to change

Reply:
Changed to `getTotalExerciseFrequency()`